src/rust/en.nana/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+#![no_std]
+use aidoku::{
+	prelude::*, error::Result, std::String, std::Vec, std::net::Request, std::net::HttpMethod,
+	Filter, Manga, MangaPageResult, Page, Chapter
+};
+
+mod parser;
+
+const BASE_URL: &str = "https://nana.my.id";
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut result: Vec<Manga> = Vec::new();
+	let mut url = String::new();
+
+	parser::get_filtered_url(filters, page, &mut url);
+	
+	let html = Request::new(url.as_str(), HttpMethod::Get).html()?;
+	let html_next_page = Request::new(url.as_str(), HttpMethod::Get).html()?;
+
+	parser::parse_search(html, &mut result);
+
+	Ok(MangaPageResult {
+		manga: result,
+		has_more: !html_next_page.select("a.paginate_button.current + a.paginate_button").array().is_empty(),
+	})
+}
+
+/// # Safety
+#[no_mangle]
+pub unsafe extern "C" fn get_manga_details(manga_rid: i32) -> i32 {
+    let manga = aidoku::std::ValueRef::new(manga_rid).as_object().unwrap();
+	let url = format!("{}/reader/{}", BASE_URL, manga.get("id").as_string().unwrap().read());
+
+	let mut img_url = String::new();
+	let mut mut_title = String::new();
+
+	if manga.get("title").as_string().is_err() && manga.get("cover").as_string().is_err() {
+		let html = match Request::new(url.as_str(), HttpMethod::Get).html() {
+			Ok(req) => req,
+			Err(_) => return -1,
+		};
+
+		mut_title.push_str(html.select("#archivePagesOverlay .spanh3reader").text().read().as_str().trim());
+
+		let img = html.select("a#display img").attr("src").read();
+		let formatted_img_url = if img.starts_with('/') {
+			format!("{}{}", BASE_URL, img)
+		} else {
+			img
+		}.replace("/image/page", "/image/thumbnails");
+
+		img_url.push_str(formatted_img_url.as_str());
+
+	}
+
```

Comment:
you can do this better (untested)
```suggestion
	let (title, cover) = if manga.get("title").as_string().is_err() && manga.get("cover").as_string().is_err() {
		let html = match Request::new(url.as_str(), HttpMethod::Get).html() {
			Ok(req) => req,
			Err(_) => return -1,
		};

		let title = html.select("#archivePagesOverlay .spanh3reader").text().read().trim();

		let img = html.select("a#display img").attr("src").read();
		let cover = if img.starts_with('/') {
			format!("{}{}", BASE_URL, img)
		} else {
			img
		}.replace("/image/page", "/image/thumbnails");

		(title, cover)
	} else {
		(manga.get("title").as_string().unwrap.read(), manga.get("cover").as_string().unwrap.read())
	}
```

Reply:
tried your method and it's working with no problems at all